### ISSUE TITLE
Please hook up api doc generation in the build system

### ISSUE THREAD

**Issue by mbiebl** [systemd/python-systemd#1]
The sphinx-based API documentation in systemd/docs/ is currently not hooked up in the build process and one needs to run spinx manually. It would be nice if the API documentation could be generated as part of the build process or at least document how one needs to run sphinx to generate the documentation. So far I failed to do that, because spinx did not find the systemd python modules.

**Comment by mbiebl** [comment]
Those are my attempts to build the api documentation:

```
[michael@pluto python-systemd (master)]$ python3 setup.py build
running build
running build_py
creating build/lib.linux-x86_64-3.4
creating build/lib.linux-x86_64-3.4/systemd
copying systemd/__init__.py -> build/lib.linux-x86_64-3.4/systemd
copying systemd/journal.py -> build/lib.linux-x86_64-3.4/systemd
copying systemd/daemon.py -> build/lib.linux-x86_64-3.4/systemd
running build_ext
building 'systemd/_journal' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/_journal.c -o build/temp.linux-x86_64-3.4/systemd/_journal.o
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/pyutil.c -o build/temp.linux-x86_64-3.4/systemd/pyutil.o
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.4/systemd/_journal.o build/temp.linux-x86_64-3.4/systemd/pyutil.o -lsystemd -o build/lib.linux-x86_64-3.4/systemd/_journal.cpython-34m.so
building 'systemd/_reader' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/_reader.c -o build/temp.linux-x86_64-3.4/systemd/_reader.o
systemd/_reader.c: In function ‘Reader_next’:
systemd/_reader.c:351:16: warning: ‘r’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         return PyBool_FromLong(r);
                ^
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/pyutil.c -o build/temp.linux-x86_64-3.4/systemd/pyutil.o
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/strv.c -o build/temp.linux-x86_64-3.4/systemd/strv.o
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.4/systemd/_reader.o build/temp.linux-x86_64-3.4/systemd/pyutil.o build/temp.linux-x86_64-3.4/systemd/strv.o -lsystemd -o build/lib.linux-x86_64-3.4/systemd/_reader.cpython-34m.so
building 'systemd/_daemon' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/_daemon.c -o build/temp.linux-x86_64-3.4/systemd/_daemon.o
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/pyutil.c -o build/temp.linux-x86_64-3.4/systemd/pyutil.o
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.4/systemd/_daemon.o build/temp.linux-x86_64-3.4/systemd/pyutil.o -lsystemd -o build/lib.linux-x86_64-3.4/systemd/_daemon.cpython-34m.so
building 'systemd/id128' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/id128.c -o build/temp.linux-x86_64-3.4/systemd/id128.o
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/pyutil.c -o build/temp.linux-x86_64-3.4/systemd/pyutil.o
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.4/systemd/id128.o build/temp.linux-x86_64-3.4/systemd/pyutil.o -lsystemd -o build/lib.linux-x86_64-3.4/systemd/id128.cpython-34m.so
building 'systemd/login' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/login.c -o build/temp.linux-x86_64-3.4/systemd/login.o
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/pyutil.c -o build/temp.linux-x86_64-3.4/systemd/pyutil.o
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DPACKAGE_VERSION="230" -I/usr/include/python3.4m -c systemd/strv.c -o build/temp.linux-x86_64-3.4/systemd/strv.o
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.4/systemd/login.o build/temp.linux-x86_64-3.4/systemd/pyutil.o build/temp.linux-x86_64-3.4/systemd/strv.o -lsystemd -o build/lib.linux-x86_64-3.4/systemd/login.cpython-34m.so
[michael@pluto python-systemd (master)]$ sphinx-
sphinx-apidoc      sphinx-autogen     sphinx-build       sphinx-quickstart  
[michael@pluto python-systemd (master)]$ sphinx-build systemd/docs/ html
Making output directory...
Running Sphinx v1.2.3
loading pickled environment... not yet created
No builder selected, using default: html
loading intersphinx inventory from http://docs.python.org/objects.inv...
building [html]: targets for 5 source files that are out of date
updating environment: 5 added, 0 changed, 0 removed
reading sources... [100%] login                                                                                         
/home/michael/debian/build-area/python-systemd/systemd/docs/daemon.rst:4: WARNING: autodoc: failed to import module u'systemd.daemon'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd.daemon
/home/michael/debian/build-area/python-systemd/systemd/docs/id128.rst:4: WARNING: autodoc: failed to import module u'systemd.id128'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd.id128
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:4: WARNING: autodoc: failed to import module u'systemd.journal'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd.journal
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:11: WARNING: don't know which module to import for autodocumenting u'JournalHandler' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:16: WARNING: don't know which module to import for autodocumenting u'_Reader' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:20: WARNING: don't know which module to import for autodocumenting u'Reader' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:26: WARNING: don't know which module to import for autodocumenting u'_get_catalog' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:27: WARNING: don't know which module to import for autodocumenting u'get_catalog' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:29: WARNING: don't know which module to import for autodocumenting u'Monotonic' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:31: WARNING: autodoc: failed to import attribute u'journal.DEFAULT_CONVERTERS' from module u'systemd'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:54: WARNING: autodoc: failed to import attribute u'journal.LOCAL_ONLY' from module u'systemd'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:55: WARNING: autodoc: failed to import attribute u'journal.RUNTIME_ONLY' from module u'systemd'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:56: WARNING: autodoc: failed to import attribute u'journal.SYSTEM' from module u'systemd'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:57: WARNING: autodoc: failed to import attribute u'journal.CURRENT_USER' from module u'systemd'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:62: WARNING: autodoc: failed to import attribute u'journal.NOP' from module u'systemd'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:63: WARNING: autodoc: failed to import attribute u'journal.APPEND' from module u'systemd'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd
/home/michael/debian/build-area/python-systemd/systemd/docs/journal.rst:64: WARNING: autodoc: failed to import attribute u'journal.INVALIDATE' from module u'systemd'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd
/home/michael/debian/build-area/python-systemd/systemd/docs/login.rst:4: WARNING: autodoc: failed to import module u'systemd.login'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
ImportError: No module named systemd.login
/home/michael/debian/build-area/python-systemd/systemd/docs/login.rst:7: WARNING: don't know which module to import for autodocumenting u'Monitor' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] login                                                                                          
writing additional files... (0 module code pages) genindex search
copying static files... done
copying extra files... done
dumping search index... done
dumping object inventory... done
build succeeded, 19 warnings.
```

**Comment by keszybz** [comment]
This was fixed in e7069690396fd22f419da13246f60e583431007f
```
make sphinx-html
```
